nmdc_schema/dump_single_modality.py

```diff
@@ -0,0 +1,342 @@
+    max_docs_per_coll = ctx.obj['MAX_DOCS_PER_COLL']
+    selected_collections = ctx.obj['SELECTED_COLLECTIONS']
+
+    (database_slots, selected_vs_schema) = fetch_and_log_schema_info(ctx)  # just printing for now
+
+    # Make requests call to fetch collection stats
+    collection_stats_url = f"{client_base_url}/{endpoint_prefix}/collection_stats"
+    response = requests.get(collection_stats_url)
+
+    documents = {}
+
+    if response.status_code == 200:
+        collection_stats = response.json()
+        logger.info("Collection statistics:")
+
+        # sort collection_stats by the ns value
+        collection_stats = sorted(collection_stats, key=lambda x: x.get("ns"))
+        for stats in collection_stats:
+            ns = stats.get("ns")
+            storage_stats = stats.get("storageStats")
+            logger.info(f"Namespace: {ns}")
+            logger.info(f"Storage Stats:")
+            logger.info(pprint.pformat(storage_stats))
+            logger.info("------------------------------------")
+            as_collection_name = ns.split(".")[-1]
+            if as_collection_name in selected_collections:
+                documents[as_collection_name] = get_collection_documents(
+                    client_base_url,
+                    as_collection_name,
+                    endpoint_prefix,
+                    max_docs_per_coll,
+                    page_size,
+                )
+                logger.info(f"{len(documents) = }")
+            else:
+                logger.warning(f"Collection '{ns}' is not selected.")
+    else:
+        logger.error(f"Failed to fetch collection stats: {response.status_code} - {response.reason}")
+
+    output_yaml = ctx.obj['OUTPUT_YAML']
+
+    logger.info(f"Writing data to {output_yaml}")
+    yaml_dumper.dump(documents, output_yaml)
+    logger.info(f"Data from PyMongo successfully written to {output_yaml}")
+
+
+@cli.command()
+@click.pass_context
+@click_log.simple_verbosity_option(logger)
+@click.option('--env-file', default='local/.env', show_default=True,
+              help='Path to .env file containing MongoDB credentials.', type=click.Path(exists=True))
+@click.option('--mongo-db-name', default="nmdc", show_default=True, help='MongoDB database name')
+@click.option('--mongo-host', default="localhost", show_default=True,
+              help='MongoDB host name/address.')
+@click.option('--mongo-port', default=27777, type=int, show_default=True, help='MongoDB port')
+@click.option('--admin-db', default=None, help='MongoDB authentication source. Leave blank to not specify.')
+@click.option('--auth-mechanism', default=None, show_default=True,
+              help='Authentication mechanism for MongoDB connection. Leave blank to not specify.')
+@click.option('--direct-connection/--no-direct-connection', default=True,
+              help='Whether to use a direct connection to MongoDB. Defaults to direct connection.')
+def pymongo_access(ctx, env_file, mongo_db_name, mongo_host, mongo_port,
```

Comment:
Another function naming comment - I think the salient point about this function is that it is directly from the DB, not mediated by the API. How it's implemented, via `pymongo` or some other means is something the user shouldn't have to be concerned with

Reply:
sorry, I see you had proposed different `cli.command` names. I went with `dump_from_api` and `dump_from_database`

Reply:
I'm going to merge now but am open to more revision if necessary.